Symfony/CS/Fixer/Contrib/PhpdocToCommentFixer.php

```diff
@@ -0,0 +1,187 @@
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'Docblocks should only be used on structural elements.';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // should be run before the PhpdocIndentFixer, PhpdocParamsFixer and NoEmptyLinesAfterPhpdocsFixer
+        return 5;
+    }
+
+    /**
+     * Check if token is a structural element
+     *
+     * @see http://www.phpdoc.org/docs/latest/glossary.html#term-structural-elements
+     *
+     * @param Token $token
+     *
+     * @return bool
+     */
+    private function isStructuralElement(Token $token)
+    {
+        $skip = array(
+            T_PRIVATE,
+            T_PROTECTED,
+            T_PUBLIC,
+            T_FUNCTION,
+            T_ABSTRACT,
+            T_CONST,
+            T_NAMESPACE,
+            T_REQUIRE,
+            T_REQUIRE_ONCE,
+            T_INCLUDE,
+            T_INCLUDE_ONCE,
+        );
+
+        return $token->isClassy() || $token->isGivenKind($skip);
+    }
+
+    /**
+     * Checks foreach statements for correct docblock usage.
+     *
+     * @param Tokens $tokens
+     * @param int    $index
+     *
+     * @return bool
+     */
+    private function isValidForeach(Tokens $tokens, $index)
+    {
+        $startIndex = $index;
+        $endIndex = $tokens->getNextTokenOfKind($startIndex, array(')'));
+        while ($startIndex < $endIndex) {
+            ++$startIndex;
+            if (!$tokens[$startIndex]->isGivenkind(T_AS)) {
+                continue;
+            }
+
+            /*
+             * Is the $value documented in docblock: foreach($array as $value)
+             * or
+             * Is the $key documented in docblock: foreach($array as $key => $value)
+             */
+            $nextMeaningful = $tokens->getNextMeaningfulToken($startIndex);
+            if (strpos($tokens[$index]->getContent(), $tokens[$nextMeaningful]->getContent()) !== false) {
+                return true;
+            }
+
+            // Is the $value documented in docblock: ($array as $key => $value)
+            $nextMeaningful = $tokens->getNextMeaningfulToken($tokens->getNextMeaningfulToken($nextMeaningful));
+            if (strpos($tokens[$index]->getContent(), $tokens[$nextMeaningful]->getContent()) !== false) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Checks variable assignments for correct docblock usage.
+     *
+     * @param Tokens $tokens
+     * @param int    $index
+     *
+     * @return bool
+     */
+    private function isValidVariable(Tokens $tokens, $index)
+    {
+        $variable = $tokens->getNextMeaningfulToken($index);
+        $nextIndex = $tokens->getNextMeaningfulToken($variable);
+        if (!$tokens[$nextIndex]->equals('=')) {
+            return false;
+        }
+
+        return strpos($tokens[$index]->getContent(), $tokens[$variable]->getContent()) !== false;
```

Comment:
nope, please see #701.
Yoda via Symfony puts const/literal on the left side, like here.
Fixer that was temporary reverted was to move all r-values on the left, even functon calls (like change `$foo === $bar->baz()` into `$bar->baz() === $foo`.